compile/service.go

```diff
@@ -207,6 +210,20 @@ func (f *FunctionSpec) Link(scope Scope) error {
 	return nil
 }
 
+// MethodName returns the method name for this function.
+func (f *FunctionSpec) MethodName() string {
+	return f.Name
+}
+
+// Type returns the envelope type that is used when making enveloped requests
+// for this function.
+func (f *FunctionSpec) Type() wire.EnvelopeType {
+	if f.OneWay {
+		return wire.OneWay
+	}
+	return wire.Call
+}
```

Comment:
Renamed to `CallType` so it doesn't implement the interface, but still gives access to the `wire.EnvelopeType`